Src/LogData.c

```diff
@@ -10,342 +19,280 @@
 #include "Data.h"
 #include "FlightPhase.h"
 
-static int SLOW_LOG_DATA_PERIOD = 700;
-static int FAST_LOG_DATA_PERIOD = 200;
-static uint8_t softwareVersion = 104;
-static uint8_t deviceAddress = 0x50;
-static uint8_t currMemAddress = 0x07;
-
-static FATFS fatfs;
-static FIL file;
-
-char fileName[32];
 
+/* Macros --------------------------------------------------------------------*/
+//CHECK: was max implemented somewhere else?
+#define max(a,b) \
+    ({ __typeof__ (a) _a = (a); \
+        __typeof__ (b) _b = (b); \
+        _a > _b ? _a : _b; })
+
+/* Structs -------------------------------------------------------------------*/
+typedef struct{
+    int32_t accelX;
+    int32_t accelY;
+    int32_t accelZ;
+    int32_t gyroX;
+    int32_t gyroY;
+    int32_t gyroZ;
+    int32_t magnetoX;
+    int32_t magnetoY;
+    int32_t magnetoZ;
+    int32_t barometerPressure;
+    int32_t barometerTemperature;
+    int32_t combustionChamberPressure;
+    int32_t oxidizerTankPressure;
+    int32_t gps_time;
+    int32_t latitude_degrees;
+    int32_t latitude_minutes;
+    int32_t longitude_degrees;
+    int32_t longitude_minutes;
+    int32_t antennaAltitude;
+    int32_t geoidAltitude;
+    int32_t altitude;
+    int32_t currentFlightPhase;
+    int32_t tick;
+} LogEntry; // logEntry holds data from AllData that is to be logged
+
+
+/* Constants -----------------------------------------------------------------*/
+static const int32_t SLOW_LOG_DATA_PERIOD_ms = 700; // logging period for slow routine
+static const int32_t FAST_LOG_DATA_PERIOD_ms = 200; // logging period for fast routine
```

Comment:
If we're using UPPERCASE_UNDERSCORE, the ms should also be uppercased?

Reply:
Yeah, I'd say so. If any units require lowercase letters (e.g. Bytes vs bits), we could just spell it out or use an abbreviation